src/contracts/core/AllocationManager.sol

```diff
@@ -0,0 +1,733 @@
+                    timestamps[i] = key;
+                } else {
+                    pendingMagnitude[i] = 0;
+                    timestamps[i] = 0;
+                }
+            }
+        }
+        return (pendingMagnitude, timestamps);
+    }
+
+    /**
+     * @notice Returns the pending deallocations of an operator for a given strategy and operatorSets.
+     * One of the assumptions here is we don't allow more than one pending deallocation for an operatorSet at a time.
+     * If that changes, we would need to change this function to return all pending deallocations for an operatorSet.
+     * @param operator the operator to get the pending deallocations for
+     * @param strategy the strategy to get the pending deallocations for
+     * @param operatorSets the operatorSets to get the pending deallocations for
+     * @return pendingMagnitudes the latest pending deallocation
+     */
+    function getPendingDeallocations(
+        address operator,
+        IStrategy strategy,
+        OperatorSet[] calldata operatorSets
+    ) external view returns (PendingFreeMagnitude[] memory) {
+        PendingFreeMagnitude[] memory pendingMagnitudes = new PendingFreeMagnitude[](operatorSets.length);
+        for (uint256 i = 0; i < operatorSets.length; ++i) {
+            uint256[] storage deallocationIndices =
+                _queuedDeallocationIndices[operator][strategy][_encodeOperatorSet(operatorSets[i])];
+
+            if (deallocationIndices.length > 0) {
+                uint256 deallocationIndex = deallocationIndices[deallocationIndices.length - 1];
+                PendingFreeMagnitude storage latestPendingMagnitude =
+                    _pendingFreeMagnitude[operator][strategy][deallocationIndex];
+                if (latestPendingMagnitude.completableTimestamp > block.timestamp) {
+                    pendingMagnitudes[i] = latestPendingMagnitude;
+                }
+            }
+        }
+        return pendingMagnitudes;
+    }
+
+    /// @dev Verify operator's signature and spend salt
+    function _verifyOperatorSignature(
+        address operator,
+        MagnitudeAllocation[] calldata allocations,
+        SignatureWithSaltAndExpiry calldata operatorSignature
+    ) internal {
+        // check the signature expiry
+        require(operatorSignature.expiry >= block.timestamp, SignatureExpired());
+        // Assert operator's signature cannot be replayed.
+        require(!avsDirectory.operatorSaltIsSpent(operator, operatorSignature.salt), SaltSpent());
+
+        bytes32 digestHash = calculateMagnitudeAllocationDigestHash(
+            operator, allocations, operatorSignature.salt, operatorSignature.expiry
+        );
+
+        // Assert operator's signature is valid.
+        EIP1271SignatureUtils.checkSignature_EIP1271(operator, digestHash, operatorSignature.signature);
+        // Spend salt.
+        operatorSaltIsSpent[operator][operatorSignature.salt] = true;
```

Comment:
should probably spend the salt prior to this (possibly) external call, to follow CEI pattern more consistently